numpy/distutils/clang_cl.py

```diff
@@ -0,0 +1,161 @@
+"""
+Build/Install NumPy on Windows with clang-cl using (powershell syntax)
+
+Short:
+python setup.py build --compiler=clang-cl install
+
+Full:
+python setup.py config --compiler=clang-cl `
+                build_clib --compiler=clang-cl `
+                build_ext --compiler=clang-cl `
+                install
+
+Develop mode:
+python setup.py build_ext -i --compiler=clang-cl develop
+
+Run tests:
+python runtests.py --compiler=clang-cl
+"""
+
+import os
+import subprocess
+from .system_info import platform_bits
+from .msvccompiler import _merge
+
+try:
+    from distutils._msvccompiler import MSVCCompiler, _find_exe
+except ImportError:
+    # Dummy to pass import test on non-Windows
+    class MSVCCompiler:
+        def __init__(self, verbose=0, dry_run=0, force=0):
+            raise NotImplementedError(
+                "ClangLC can only be run on Windows when MSVCCompiler can "
+                "be imported from distutils._msvccompiler"
+            )
+
+
+class ClangCL(MSVCCompiler):
+    """
+    A modified clang-cl compiler compatible with an MSVC-built Python.
+    """
+
+    compiler_type = "clang_cl"
```

Comment:
Not sure if it matters, but upstream patch seems to set this to `clang-cl`?
https://github.com/python/cpython/pull/18371/files#diff-d962ba8e352edf080a341706a50c13b0R570

Reply:
I have renamed to clang-cl for simplicity.  I think it can be anything, the string is only used in choosing paths.